javaslang/src/main/java/javaslang/collection/Array.java

```diff
@@ -546,24 +558,22 @@ public T get(int index) {
     public Array<T> drop(long n) {
         if (n <= 0) {
             return this;
-        }
-        if (n >= length()) {
+        } else if (n >= length()) {
             return empty();
+        } else {
+            return with(startIndex + (int) n, length - (int) n);
```

Comment:
the cast does not work for big `n`.
It is confusing that we use both `length()` and `length`. We should use always `length` because is an accessible instance variable.